### PR TITLE
highlight joints in the animation editor viewport when they are hover…

### PR DESCRIPTION
**Pull request by greerdv**
…ed over in the skeleton outliner

Signed-off-by: greerdv <greerdv@amazon.com>

## What does this PR do?
Highlight joints in the animation editor viewport when they are hovered over in the skeleton outliner.

## How was this PR tested?
Manual testing in the animation editor.

### CODE REVIEW

**Comment by greerdv** [comment]
I think this is basically complete in terms of code, just need to finalize the hover colour with UX (the blue here is just a placeholder).

https://user-images.githubusercontent.com/82226198/186443390-5d30309b-7318-498b-a453-b9049503cd7b.mp4

**Review by jonawals — APPROVED** [review]
LGTM

File: Code/Framework/AzFramework/AzFramework/Physics/DebugDraw/CharacterPhysicsDebugDraw.cpp [hunk 1]
```diff
@@ -44,9 +44,23 @@ namespace Physics
...
+                colliderColor = colorSettings.m_selectedColor;
+            }
+            else if ((1 << colliderIndex) & invalidColliderBitArray)
+            {
+                colliderColor = colorSettings.m_errorColor;
+            }
+            else
+            {
+                colliderColor = colorSettings.m_defaultColor;
+            }
```
Comment (hultonha) [inline Code/Framework/AzFramework/AzFramework/Physics/DebugDraw/CharacterPhysicsDebugDraw.cpp:63]: suggestion (feel free to ignore): You could move this to an immediately invoked lambda so `colliderColor` can be `const`. This nicely separates initialization from other uses.

```
const AZ::Color colliderColor = [] {
    if (nodeDebugDrawData.m_hovered)
    {
        return colorSettings.m_hoveredColor;
    }
    else if (nodeDebugDrawData.m_selected)
    {
        return olorSettings.m_selectedColor;
    }
    else if ((1 << colliderIndex) & invalidColliderBitArray)
    {
        return colorSettings.m_errorColor;
    }
    return colorSettings.m_defaultColor;
}();
```

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 2]
```diff
@@ -229,34 +232,45 @@ namespace AZ::Render
         EMotionFX::JointSelectionRequestBus::BroadcastResult(
             cachedSelectedJointIndices, &EMotionFX::JointSelectionRequests::FindSelectedJointIndices, instance);
 
+        size_t cachedHoveredJointIndex;
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:235]: update: We should initialize `cachedHoveredJointIndex` to `0`

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 3]
```diff
@@ -266,7 +280,9 @@ namespace AZ::Render
                 debugDisplay,
                 instance->GetActor()->GetPhysicsSetup()->GetColliderConfigByType(EMotionFX::PhysicsSetup::SimulatedObjectCollider),
                 nodeDebugDrawDataFunction,
-                { renderActorSettings.m_simulatedObjectColliderColor, renderActorSettings.m_selectedSimulatedObjectColliderColor });
+                { renderActorSettings.m_simulatedObjectColliderColor,
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:283]: suggestion: This is minor and not 100% directly related to your change, but can you add the type of the initializer here for the braced initializer - it makes tracking down type usages so much easier in future (trust me, I know 😅)

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 4]
```diff
@@ -294,15 +310,24 @@ namespace AZ::Render
...
 
             m_characterPhysicsDebugDraw.RenderRagdollColliders(
-                                debugDisplay, ragdollColliderConfiguration, nodeDebugDrawDataFunction, parentIndices,
-                                { renderActorSettings.m_ragdollColliderColor, renderActorSettings.m_selectedRagdollColliderColor,
-                                  renderActorSettings.m_violatedRagdollColliderColor });
+                debugDisplay,
+                ragdollColliderConfiguration,
+                nodeDebugDrawDataFunction,
+                parentIndices,
+                { renderActorSettings.m_ragdollColliderColor,
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:317]: suggestion: Same comment about adding type for object created here just before the braces

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 5]
```diff
@@ -427,6 +472,10 @@ namespace AZ::Render
         EMotionFX::JointSelectionRequestBus::BroadcastResult(
             cachedSelectedJointIndices, &EMotionFX::JointSelectionRequests::FindSelectedJointIndices, instance);
 
+        size_t cachedHoveredJointIndex;
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:475]: update: `cachedHoveredJointIndex` should be initialized to `0` (or some invalid value we can check against if the bus call fails for some reason)

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp [hunk 6]
```diff
@@ -475,6 +517,10 @@ namespace AZ::Render
...
```
Comment (hultonha) [inline Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:520]: update: Same comment as above

**Review by hultonha — APPROVED** [review]
Change looks good just a few small suggestions that would be worth making before merging

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 2 above)
Comment (hultonha) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:235]: Apologies this should more likely be `InvalidIndex`, but we should initialize it to something 😅

Comment (greerdv) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:235]: oh yeah, good call on initializing it

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 3 above)
Comment (greerdv) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:283]: Good idea, done!

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 4 above)
Comment (greerdv) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:317]: Done

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 5 above)
Comment (greerdv) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:475]: Fixed

File: Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp (hunk 6 above)
Comment (greerdv) [reply Gems/AtomLyIntegration/EMotionFXAtom/Code/Source/AtomActorDebugDraw.cpp:520]: Same fix as above :p

File: Code/Framework/AzFramework/AzFramework/Physics/DebugDraw/CharacterPhysicsDebugDraw.cpp (hunk 1 above)
Comment (greerdv) [reply Code/Framework/AzFramework/AzFramework/Physics/DebugDraw/CharacterPhysicsDebugDraw.cpp:63]: I tried that out, but I had to capture so many things in the lambda that it felt a bit clunky.

**Comment by greerdv** [comment]
With updated colours after UX discussion
https://user-images.githubusercontent.com/82226198/186751118-9faef971-29dc-481c-8a03-39b565008d8b.mp4

**Pull request merged**